docs/src/add-services/varnish.md

```diff
@@ -97,57 +127,114 @@ sub vcl_recv {
 
 {{< readFile file="src/registry/images/examples/full/varnish.routes.yaml" highlight="yaml" location=".platform/routes.yaml" >}}
 
-That will map all incoming requests to the Varnish service rather than the application.
-Varnish will then, based on the VCL file, forward requests to the application as appropriate.
+Varnish forwards requests to your app based on the specified VCL template.
 
-## Modules
+## Include modules
 
-Platform.sh supports a number of optional modules you can include in your VCLs, namely:
+You can include the following optional modules in your VCL templates to add additional features:
 
-* `cookie`
-* `header`
-* `saintmode`
-* `softpurge`
-* `tcp`
-* `var`
-* `vsthrottle`
-* `xkey`
+- `cookie`
+- `header`
+- `saintmode`
+- `softpurge`
+- `tcp`
+- `var`
+- `vsthrottle`
+- `xkey`
 
-To use in your VCL, add an import such as:
+To use them, add an import to your template such as the following:
 
 ```bash {location=".platform/config.vcl"}
 import xkey;
 ```
 
 ## Circular relationships
 
-At this time Platform.sh doesn't support circular relationships between services or applications.
-That means you cannot add a relationship in your `.platform.app.yaml` that points to the Varnish service.
+At this time, Platform.sh doesn't support circular relationships between services and apps.
+That means you can't add a relationship from an app fronted by Varnish to the Varnish service.
 If you do so, then one of the relationships is skipped and the connection doesn't work.
-This limitation may be lifted in the future.
+
+## Rate limit connections
+
+Sometimes you want to ensure that users, whether human or machine, can't overload your app with requests.
+One tool to help is using Varnish to limit the rate at which connections can be made.
+
+For example, say you want to make sure no one can make more than 20 requests within 10 seconds.
+If they do, you want to block them from any more requests for 2 minutes.
+
+To do so, [import the `vsthrottle` module](#include-modules)
+and add logic similar to the following to your VCL template:
+
+```bash {location=".platform/config.vcl"}
+import vsthrottle;
+
+sub vcl_recv {
+  # The Platform.sh router will provide the real client IP as X-Client-IP
+  # This replaces client.identity in other implementations
+  if (vsthrottle.is_denied(req.http.X-Client-IP, 20, 10s, 120s)) {
+    # Client has exceeded 20 requests in 10 seconds.
+    # When this happens, block that IP for the next 120 seconds.
+    return (synth(429, "Too Many Requests"));
+  }
+  
+  # Set the standard backend for handling requests that aren't limited
+  set req.backend_hint = application.backend();
+}
+```
+
+## Clear cache with a push
+
+You may want at times to clear a specific part of your cache when you know the content is outdated.
+With Varnish, you can clear the content with [purging and banning](https://varnish-cache.org/docs/trunk/users-guide/purging.html).
+
+The following example shows how to set up purging.
```

Comment:
:hammer_and_wrench:

Based on the previous comment, would the following:
```suggestion
The following example shows how to set up purging from your app.
```
Help make the context a bit clearer?

Reply:
The example doesn't really show that. But I can add clarification.